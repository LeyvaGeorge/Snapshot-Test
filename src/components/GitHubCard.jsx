import { Card } from "react-bootstrap"

export default function GitHubCard() {
    return(
        <div>
        </div>
    )
}