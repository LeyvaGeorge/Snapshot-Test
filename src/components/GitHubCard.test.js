import { render, screen, waitFor } from "@testing-library/react";
import renderer from "react-test-renderer";
import GitHubCard from "./githubCard";

const DEFAULT_MOCK_DATA = {
    name: 'alpha one',
    html_url: 'https://placekitten.com/200/300',
    bio:'crazy'
}

describe('Test GitHubCard',() => {
    beforeEach(() => {
        fetch.mockResponseOnce(JSON.stringify(DEFAULT_MOCK_DATA));
        render(<GitHubCard/>);
    })
    afterEach(() => {
        fetch.resetMocks();
    })
    test('it contains a photo of the github user', async () => {
        const avatarImage = await waitFor(() => screen.getAllByAltText('git by avatar'));
        expect(avatarImage).toHaveAttribute('src',
        expect.stringContaining(DEFAULT_MOCK_DATA.avatar_url)
        );
    });

    test('it contains the name of the github user', async() => {
        const userName = await waitFor(() => screen.getByRole('heading', { level: 2 }));
        expect(userName).toHaveTextContent(DEFAULT_MOCK_DATA.name)
    });

    test('it contains the blurb about the github user', async() => {
        const blurb = await waitFor(() => screen.getByRole('paragraph'));
        expect(blurb).toHaveTextContent(DEFAULT_MOCK_DATA.bio)
    });

    test('it renders a snapshot', () => {
        const tree = renderer.create(<GitHubCard/>).toJSON();
        expect(tree).toMatchSnapshot();
    });


})