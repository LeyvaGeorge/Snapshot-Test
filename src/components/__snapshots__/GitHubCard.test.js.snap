// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test GitHubCard it renders a snapshot 1`] = `<div />`;
